### PR TITLE
feat: ledger compatability

### PR DESCRIPTION
**Pull request by SWvheerden**
This PR is to add no_std support. This is in-line to ensure that tari_utilities can compile for nonstandard tier 1 rust targets. This is required for the push to ensure we can create a ledger-compatible library and allow us to run some of the handy utilities from this crate. 

The changes can be summarized in the following:

- Made the library `no_std`
- Upgraded borsch to version 0.10, up from 0.9
- Added feature flags to hide crates and features that are not possible on `no_std` targets

Feature flags added:

- `std` - This will allow epoch_time, encoding and safe_array
- `zero` - This will include zeroize support
- `serialize` - This will include support for serde
- `borsh` - This will include support for borsh
- `default` - This will include all feature flags and allow all utilities.

### CODE REVIEW

**Review by hansieodendaal — APPROVED** [review]
ACK

**Review by stringhandler — COMMENTED** [review]
Happy in concept, but needs the following before merging:

1. An updated readme describing what each of the features are for and when to use them
2. Better PR description explaining the changes and motivation of choices
3. A CI job to check that each feature compiles correctly. I.E. `cargo check --feature monero`. When there are many features on a crate it is easy for it to have some subset of features fail compilation, so we need to check these build for this and future PRs
4. A CI job specifically for no-std if that's what you're aiming to achieve with this PR

File: README.md [hunk 1]
```diff
@@ -54,4 +54,30 @@ A `MessageFormat` trait that handles conversion from and to binary, json, or bas
 
 ## Hidden
 
-A wrapper type for concealing sensitive information in logs.
\ No newline at end of file
+A wrapper type for concealing sensitive information in logs.
+
```
Comment (stringhandler) [inline README.md:58]:
```suggestion
# Feature Flags
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -44,12 +44,44 @@ jobs:
...
         uses: actions-rs/cargo@v1
         with:
           command: build
-          args: --release
+          args: --release --all-features
+      - name: cargo build std
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --release --features "std"
```
Comment (stringhandler) [inline .github/workflows/test.yml:64]:
```suggestion
          args: --release --no-default-features --features "std"
```

**Review by stringhandler — COMMENTED** [review]
Much better, just need to ensure the ci jobs run only those features, and not the default features and the specific feature. I think the correct option is `--no-default-features`

**Review by stringhandler — APPROVED** [review]

**Pull request merged**